User-Data Store Migration/dump-converter.py

```diff
@@ -138,26 +139,32 @@ def writeline(stream, line):
 def initialize(out):
     global pre, replace_rules, post, NAME_TEMPLATE
 
+    # Delete any action on the internal sqlite_sequence table
     replace_rules.append((True, r"^.*sqlite_sequence.*;[\r\n]*$", "", 0, None, True))
-    replace_rules.append((True, r"^PRAGMA foreign_keys=OFF;[\r\n]*$", "", 0, None, True))
+    # Fix the datetime values
     replace_rules.append((False, " +00:00'", "'", 0, None, False))
-    replace_rules.append((True, r"^INSERT INTO.*$", None, 0, injectStructureInfoInsertInto, False))
+    # Rewrite compatible insert rules
+    replace_rules.append((True, r"^INSERT INTO.*$", None, 0, injectStructureInfoInsertInto, True))
 
     if __dialect__ in ["mysql", "mariadb"]:
         NAME_TEMPLATE = "`%s`"
+        pre.append("START TRANSACTION;")
+        # Disable/enable the constraint checks
         pre.append("SET FOREIGN_KEY_CHECKS=0;")
         post.append("SET FOREIGN_KEY_CHECKS=1;")
         
-        replace_rules.append((True, r"^BEGIN TRANSACTION;\s*$", "START TRANSACTION;", 0, None, True))
+        # Extract the table structure and replace the CREATE statement with a TRUNCATE
         replace_rules.append((True, r"^CREATE TABLE `(?P<name>[^`]*)`.*$", ("TRUNCATE TABLE " + NAME_TEMPLATE + ";") % ("{1}"), 1, getStructureInfoCreate, True))
     elif __dialect__ == "mssql":
         NAME_TEMPLATE = "[%s]"
+        pre.append("BEGIN TRANSACTION;")
         pre.append("declare @query varchar(max);");
+        # Disable/enable the constraint checks for every table
         pre.append("select @query = coalesce(@query + ' ' + 'ALTER TABLE [' + name + '] NOCHECK CONSTRAINT all;', 'ALTER TABLE [' + name + '] NOCHECK CONSTRAINT all;') from sys.tables; exec(@query);")
         post.append("select @query = coalesce(@query + ' ' + 'ALTER TABLE [' + name + '] WITH CHECK CHECK CONSTRAINT all;', 'ALTER TABLE [' + name + '] WITH CHECK CHECK CONSTRAINT all;') from sys.tables; exec(@query);")
-        post.append("go")

```

Comment:
how is the generated script affected by moving "go" from here to the finalize  function?

Reply:
I need the `go`  to be the last line to be sure that all the tools interpret it as a script to be executed. Before I was keeping the `commit` from the input, to the `post` was writing everything at the end of the file.

with the new changes (see other comments) I explicitly write the `commit` at the end of the file and I need to put the `go` after that.

Notice that as implicit change now I have the start transaction as very first line of the file and the commit as very last one. Before I had some other lines before the start or after the commit that weren't really impacted by the transaction it self.